### PR TITLE
Password criteria in new user registration form

### PR DESCRIPTION
**Pull request by byewokko**
follow-up from #43

# Changes
- New user registration form now also uses advanced password criteria loaded from the backend.
- Created a new component `PasswordChangeFieldGroup` that is reused in _Change password screen_, _Reset password screen_ and _New user registration screen_.

### CODE REVIEW

File: public/locales/cs/translation.json [hunk 1]
```diff
@@ -47,11 +47,23 @@
 		"Invalid format, only lower-case letters, numbers, dash and underscore are allowed": "Nesprávný formát, pouze malá písmena, číslice, pomlčka a podtržítko jsou povoleny",
 		"Only lower-case letters, numbers, dash and underscore are allowed": "Pouze malá písmena, číslice, pomlčka a podtržítko jsou povoleny",
 		"Username editing is not allowed": "Není povoleno upravit uživatelské jméno",
-		"Password": "Heslo",
-		"Re-enter Password": "Heslo znovu",
-		"Password can't be empty!": "Heslo nemůže být prázdné!",
-		"Passwords do not match!": "Hesla se musí shodovat!",
-		"Required field": "Povinný údaj"
+		"Required field": "Povinný údaj",
+		"PasswordChange": {
```
Comment (Pe5h4) [inline public/locales/cs/translation.json]: @byewokko We use 2 level nesting only in some special cases for translations (or we did in the very first implementation of localization - so you may still find it somewhere). I know its possible, but I would advice to create a separate translation key, than to make a nested out of it.

**Review by Pe5h4 — COMMENTED** [review]
Seems OK-ish, please reconsider the localization key refactorisation

File: public/locales/cs/translation.json (hunk 1 above)
Comment (byewokko) [reply public/locales/cs/translation.json]: flattened :heavy_check_mark:

File: src/modules/auth/containers/FormFields.js [hunk 2]
```diff
@@ -127,72 +135,140 @@ export function UserNameField(props) {
...
+				name='newpassword'
+				type='password'
+				autoComplete='new-password'
+				required='required'
+				invalid={Boolean(errors?.newpassword)}
+				onBlur={regNewPassword.onBlur}
+				innerRef={regNewPassword.ref}
+				onChange={regNewPassword.onChange}
+			/>
+			{errors?.newpassword?.type !== 'passwordCriteria'
```
Comment (Pe5h4) [inline src/modules/auth/containers/FormFields.js]: @byewokko Ideally you should wrap the condition into parenthesis:

```
(errors?.newpassword?.type !== 'passwordCriteria')
```

Myabe we should add this into eslint criteria

**Review by Pe5h4 — APPROVED** [review]

File: src/modules/auth/containers/FormFields.js (hunk 2 above)
Comment (byewokko) [reply src/modules/auth/containers/FormFields.js]: we dont have eslint active for this repository yet

i'll fix this one

**Pull request merged**